File: python_solutions/boj_3986.py
import sys
input = sys.stdin.readline

n = int(input())
table = [list(input().strip()) for _ in range(n)]
cnt = 0

for word in table:
    stack = []
    top = -1
    for a in word:
        if not stack:
            stack.append(a)
            top += 1
        else:
            if stack[top] == a:
                stack.pop()
                top -= 1
            else:
                stack.append(a)
                top += 1
    if not stack:
        cnt += 1

print(cnt)

File: cpp_solutions/boj_3986/boj_3986.cpp
#include <iostream>
#include <string>
#include <stack>

using namespace std;

int main(){
    int cnt = 0, N; cin >> N;

    while (N--){
        string ss;
        cin >> ss;
        
        if (ss.length() % 2 != 0) continue;

        stack<char> st;

        for(char c : ss){
            if (!st.empty() && st.top() == c) st.pop();
            else st.push(c);
        }
        if (st.empty()) cnt++;
    }

    cout << cnt;
}